src/backend/joanie/core/api.py

```diff
@@ -196,16 +196,18 @@ def create(self, request, *args, **kwargs):
         return Response(serializer.data, status=201)
 
     @action(detail=True, methods=["POST"])
-    def abort(self, request, uid=None):  # pylint: disable=no-self-use
+    def abort(
+        self, request, id=None
+    ):  # pylint: disable=no-self-use,invalid-name, redefined-builtin
         """Abort a pending order and the related payment if there is one."""
         username = request.user.username
         payment_id = request.data.get("payment_id")
 
         try:
-            order = models.Order.objects.get(uid=uid, owner__username=username)
+            order = models.Order.objects.get(id=id, owner__username=username)
```

Comment:
Related to my previous comment.
```suggestion
            order = models.Order.objects.get(pk=pk, owner__username=username)
```